### PR TITLE
Add event system revised

### PR DESCRIPTION
**Pull request by sriking1783**
Can we still submit PR's?

### CODE REVIEW

**Comment by ashtonthomas** [comment]
@sriking1783, yeah you can still submit PR's!

File: event_system/db/seeds.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
```
Comment (djkotowski) [inline event_system/db/seeds.rb]: If you're not using it, delete it.

File: event_system/features/step_definitions/weather_steps.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+Given(/^I visit "(.*?)" page$/) do |arg1|
```
Comment (djkotowski) [inline event_system/features/step_definitions/weather_steps.rb]: Rename cuke params to improve clarity.

File: event_system/features/step_definitions/weather_steps.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  visit arg1
+end
+
+Given(/^I submit the city name$/) do
+  find(:css, "#city_name").set("Chicago")
```
Comment (djkotowski) [inline event_system/features/step_definitions/weather_steps.rb]: You could also use `fill_in` here.
```ruby
fill_in("#city_name", with: "Chicago")
```

File: event_system/features/step_definitions/weather_steps.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  find(:link_or_button, "Send").click
+end
+
+Given(/^I did not submit the city name$/) do
+  find(:css, "#city_name").set("Chicago")
+end
+
+Then(/^I should see temperature forecast$/) do
+  expect(page).to have_selector('#temperatue_forecast_table')
```
Comment (djkotowski) [inline event_system/features/step_definitions/weather_steps.rb]: Spelling.

File: event_system/features/support/env.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+#
+ActionController::Base.allow_rescue = false
+
+# Remove/comment out the lines below if your app doesn't have a database.
+# For some databases (like MongoDB and CouchDB) you may need to use :truncation instead.
+begin
+  DatabaseCleaner.strategy = :transaction
+rescue NameError
+  raise "You need to add database_cleaner to your Gemfile (in the :test group) if you wish to use it."
+end
```
Comment (djkotowski) [inline event_system/features/support/env.rb]: Database Cleaner is very useful.  You can set it up to give you a clean slate after each scenario.

File: event_system/features/index.feature [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+  Scenario: Should not see high low temperatures
+    Given I visit "weather/index" page
+    And I did not submit the city name
+    Then I should not see temperature forecast
+
+  Scenario: Should see high low temperatures
+    Given I visit "weather/index" page
+    And I submit the city name
+    Then I should see temperature forecast
```
Comment (djkotowski) [inline event_system/features/index.feature]: What about sad/bad paths?  Invalid cities?  Submitting blanks?

File: event_system/lib/web_scraper.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
+require 'nokogiri'
+class WebScraper
+  def self.scrape_temperatures(doc)
+    document = doc
```
Comment (djkotowski) [inline event_system/lib/web_scraper.rb]: Why the reassignment?

File: event_system/lib/web_scraper.rb [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        low_high_temp[date]["low"] = para.css("p.point-forecast-icons-low").text
+        .split(" ")[1]
+        date += 1.day
+      else
+        low_high_temp[date] ||= {}
+        low_high_temp[date]["high"] = para.css("p.point-forecast-icons-high").text
+        .split(" ")[1]
+      end
+    end
+    low_high_temp
```
Comment (djkotowski) [inline event_system/lib/web_scraper.rb]: You can get around these steps with `#each_with_object`.

```ruby
date = Date.today
document.css("li.forecast-tombstone").each_with_object({}) do |para, low_high_temp|
  # your code here
end
```

File: event_system/lib/web_scraper.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def self.detailed_scrape(doc)
+    document = doc
+    forecast_description = {}
```
Comment (djkotowski) [inline event_system/lib/web_scraper.rb]: Again, you can use `#each_with_object` here.

File: event_system/lib/web_scraper.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    top_level = document.css('div#detailed-forecast-body')
+    date = Date.today
+    document.css('div.row-forecast').each do |para|
+      temp_key = para.css('div.forecast-label').text
+        if temp_key[-5..-1].downcase == "night"
+          forecast_description[date] ||= {}
+          forecast_description[date]["detail_night"] = para.css('div.forecast-text').text.strip
+          date += 1.day
+        else
+          forecast_description[date] ||= {}
```
Comment (djkotowski) [inline event_system/lib/web_scraper.rb]: This step is repeated in both branches of the `if` statement.  Extract it out.

File: event_system/spec/controllers/register_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,21 @@
+require 'spec_helper'
+require 'rspec/rails'
+
+describe RegisterController, type: :controller do
+
+  describe "index" do
+  end
```
Comment (djkotowski) [inline event_system/spec/controllers/register_controller_spec.rb]: Remove unused descriptions.

File: event_system/spec/controllers/weather_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+  context "GET index" do
+
+  end
+
+  context "GET show" do
+
+  end
+
+end
```
Comment (djkotowski) [inline event_system/spec/controllers/weather_controller_spec.rb]: No need for unused specs.  Delete them.

File: event_system/spec/interactors/data_repo_spec.rb [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+      expect(DataRepo).to receive(:parse_scraped_data).and_return(Proc.new{})
+
+      DataRepo.call(model_name, method_name, attributes)
+    end
+
+    it "does not calls parse_scraped_data method" do
+      allow(model_name).to receive(:today).and_return([{high: 40, low: 30}])
+
+      expect(DataRepo).to receive(:structure_data).and_return(Proc.new{})
```
Comment (djkotowski) [inline event_system/spec/interactors/data_repo_spec.rb]: The test description implies you're testing
```ruby
expect(DataRepo).not_to receive(:parse_scraped_date)
```

File: event_system/spec/lib/web_scraper_spec.rb [hunk 14]
```diff
@@ -0,0 +1,18 @@
+require 'spec_helper'
+
+describe WebScraper do
+  let(:file) { File.open("temperatures.html", "r") }
+
+  it "scrapes temperature given a nodeset" do
+    doc = Nokogiri::HTML(file)
```
Comment (djkotowski) [inline event_system/spec/lib/web_scraper_spec.rb]: This should be in a `let`.

File: event_system/app/interactors/data_repo.rb [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+      scraping = forecast.call(method_name)
+    else
+      forecast = model_name.today.as_json
+      details = structure_data(attributes)
+      scraping = details.call(forecast)
+    end
+    scraping
+  end
+
+  def self.parse_scraped_data(model_name)
```
Comment (djkotowski) [inline event_system/app/interactors/data_repo.rb]: Method name implies you're parsing data, not returning a proc.

File: event_system/app/interactors/data_repo.rb [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    return Proc.new  do |method|
+      scraping = WebScraper.send method, document
+      scraping.each do |day, scraped_data|
+        save_scraped_data(day, scraped_data, model_name.new)
+      end
+    end
+  end
+
+  def self.structure_data(key_params)
```
Comment (djkotowski) [inline event_system/app/interactors/data_repo.rb]: Method name implies you're parsing data, not returning a proc.

File: event_system/app/interactors/data_repo.rb [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+    return Proc.new do |model|
+      model.each_with_object({}) do |hash, return_hash|
```
Comment (djkotowski) [inline event_system/app/interactors/data_repo.rb]: Avoid naming variables by their data types.

File: event_system/app/views/register/index.html.haml [hunk 18]
```diff
@@ -0,0 +1,6 @@
+%h1 Register#email
```
Comment (djkotowski) [inline event_system/app/views/register/index.html.haml]: Leftover title from generators.  These should be cleaned up.

**Comment by djkotowski** [comment]
Great job!  There's a lot of unused files or generated comments that you could do without.  Otherwise, though, this looks pretty clean.  No need for a resubmission.

**Pull request closed without merging**